test/e2e/throughputanomalydetection_test.go

```diff
@@ -252,8 +321,28 @@ func waitTADJobComplete(t *testing.T, data *TestData, jobName string, timeout ti
 	return nil
 }
 
-func tadrunJob(t *testing.T, data *TestData, algotype string) (stdout string, jobName string, err error) {
+func tadrunJob(t *testing.T, data *TestData, algotype, agg_type string) (stdout string, jobName string, err error) {
+	mutex.Lock()
+	defer mutex.Unlock()
 	newjobcmd := tadstartCmd + " --algo " + algotype + " --driver-memory 1G --start-time 2022-08-11T06:26:50 --end-time 2022-08-12T08:26:54"
+	switch agg_type {
+	case "podName":
+		agg_flow_ext := " --agg-flow pod"
+		ext := " --pod-name test_podName"
+		newjobcmd = newjobcmd + agg_flow_ext + ext
+	case "podLabel":
+		agg_flow_ext := " --agg-flow pod"
+		ext := " --pod-label \"test_key\":\"test_value\""
+		newjobcmd = newjobcmd + agg_flow_ext + ext
+	case "external":
+		agg_flow_ext := fmt.Sprintf(" --agg-flow %s", agg_type)
+		ext := " --external-ip 10.10.1.33"
+		newjobcmd = newjobcmd + agg_flow_ext + ext
+	case "svc":
+		agg_flow_ext := fmt.Sprintf(" --agg-flow %s", agg_type)
+		ext := " --svc-port-name test_serviceportname"
+		newjobcmd = newjobcmd + agg_flow_ext + ext
```

Comment:
`newjobcmd = newjobcmd + agg_flow_ext + ext` is same for all cases, consider moving it out of swith..case..